draft/XXXX-multinetjs-autopublish/README.md

```diff
@@ -0,0 +1,86 @@
+---
+RFC: XXXX
+Author: Roni Choudhury
+Status: draft
+Created: 2020-07-09
+Last-Modified: 2020-07-09
+---
+
+# RFC XXXX: Autopublishing for MultinetJS
+
+One of the active pieces of software driving the Multinet project is
+`multinetjs`, a TypeScript library providing clientside bindings and utility
+functions for the Multinet API. This software grows with the needs of the
+project, specifically of the [Multinet
+client](https://github.com/multinet-app/multinet-client). This means that often,
+a new feature in the client requires a new feature in MultinetJS, and to support
+this use case, this RFC lays out a plan for more easily publishing prerelease
+and development versions of the library for use in client development.
+
+## Motivation
+
+MultinetJS does not currently have an automated, official release process.
+Releases are done by hand after a pull request is approved, and before that step
+happens, developers are expected to use `yarn link` to provide development
+linkages between MultinetJS and the client codebase.
+
+While this procedure works well in local development, it leaves CI out. In
+particular, since the Netlify build previews of the client rely on published
+versions of MultinetJS, build previews depending on a development version of
+MultinetJS simply cannot be built.
+
+A procedure for automated or semi-automated publishing of MultinetJS, even for
+development versions, would fix this issue. It would also eliminate the manual
+step currently needed to perform releases to NPM.
+
+## Proposals
+
+The proposals here use GitHub Actions to drive publishing pipelines to NPM on
+both pull request branches, and on master. Details for each follow.
+
+### Development Releases
+
+The idea here is to automatically publish development versions based on a PR
+branch to NPM automatically using CI. On each push from a PR branch, the GH
+Actions workflow can run the tests, perform a build, and if successful, publish
+a prerelease version to NPM. The version number can be taken from the current
+value in `package.json`, appended with the first six characters of the current
+commit's Git SHA--for example `multinet-0.47.3-a1c0ef` ([Clause
```

Comment:
Do yarn and npm know that a version with alphanumeric appended to the end are not official releases that should be install-able unless explicitly requested? For example, in the case the 0.47.3 is the latest published version that we're supporting and 0.47.3-a1c0ef is the testing release, doing a `yarn install multinetjs` should install 0.47.3, not 0.47.3-a1c0ef

Reply:
This article is pretty old, but addresses your question (which is an excellent one): https://medium.com/@mbostock/prereleases-and-npm-e778fc5e2420.

We would need to do some experimentation to see how this works but the main idea is that a bare `yarn install multinet` will install whatever the latest published version is, but `yarn install multinet@0.47` would install `0.47.3`, not a prerelease version. Prerelease versions need to be installed explicitly, which would be the use case for Multinet client.

Reply:
Yeah, looks like the pre-release is tagged as latest so doing a `yarn add multinet` would give us the pre-release, but doing `yarn add multinet@0.47` would give the desired behavior. We could also consider using tags for pre-releases so they're not included in the glob when using `yarn add multinet` I'm interested to see the results of the experimentation and I'm glad it's on our radar. Thanks for the article @waxlamp